File: legacy-repo/src/main/java/com/clinic/legacy/AuditLogger.java
package com.clinic.legacy;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/**
 * Simple audit logger for tracking access to patient records.
 */
public class AuditLogger {

    private static final DateTimeFormatter AUDIT_DATE_FORMAT =
            DateTimeFormatter.ofPattern("M/d/yyyy");

    /**
     * Logs an access event for a given patient.
     *
     * SEEDED ISSUE (GDPR-005): logs the FULL patient record (including SSN
     * and diagnosis) to plain console/text output for every read access,
     * rather than a minimal identifier. Violates data-minimization
     * principles and creates an unencrypted trail of sensitive data.
     */
    public void logAccess(Patient patient, String accessedBy) {
        String timestamp = LocalDate.now().format(AUDIT_DATE_FORMAT);
        System.out.println("[AUDIT] " + timestamp + " - " + accessedBy
                + " accessed record: " + patient.toString());
    }

    public void logError(String message, Exception e) {
        System.out.println("[ERROR] " + message + ": " + e.getMessage());
    }
}

File: legacy-repo-python/clinic/audit_logger.py
"""
Access audit logger for the Clinic Records system.
"""
import datetime


class AuditLogger(object):

    def log_access(self, patient, accessed_by):
        """Log a patient record access to stdout.

        SEEDED ISSUE (GDPR-005): logs the full patient record including
        SSN and diagnosis on every access. Data minimization requires
        logging only the patient ID, not the full PHI payload.

        LEGACY: uses print statement (Python 2) instead of print() function.
        """
        # LEGACY: print statement — SyntaxError in Python 3.
        # Modernizer target: replace with print()
        print "[AUDIT] %s accessed by %s at %s" % (
            repr(patient),       # calls Patient.__repr__() — leaks SSN + diagnosis
            accessed_by,
            datetime.date.today().isoformat(),
        )
